# Hot Tips: 2016-08-24

## PyCharm

I should have demoed in PyCharm.
Please, still use it.

### Running Doctests

First, [follow the instructions](/notes/django-init.md) I've written up to add a `tests.py` to your Django app.
Then, you can run tests through Django.

* Script: `manage.py`
* Script parameters: `test`

You might get a warning when debugging tests:

```
PYDEV DEBUGGER WARNING:
sys.settrace() should not be used when the debugger is being used.
This may cause the debugger to stop working correctly.
If this is needed, please check:
http://pydev.blogspot.com/2007/06/why-cant-pydev-debugger-work-with.html
to see how to restore the debug tracing back correctly.
Call Location:
  File "/usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/doctest.py", line 1471, in run
    sys.settrace(save_trace)
```

That's okay.
It doesn't seem to have an effect.

### Debug Run Server

* Script: `manage.py`
* Script parameters: `runserver`
